src/promptflow/promptflow/_sdk/_telemetry/logging_handler.py

```diff
@@ -73,71 +53,127 @@ def get_scrubbed_cloud_role():
     return cloud_role
 
 
+class PromptFlowSDKExporter(AzureMonitorLogExporter):
+    def __init__(self, custom_dimensions, **kwargs):
+        super().__init__(**kwargs)
+        self._custom_dimensions = custom_dimensions
+
+    def _log_to_envelope(self, log_data: LogData) -> TelemetryItem:
+        log_data.log_record.attributes.update(self._custom_dimensions)
+        envelope = super()._log_to_envelope(log_data=log_data)
+        # scrub data before sending to appinsights
+        role = get_scrubbed_cloud_role()
+        envelope.tags["ai.cloud.role"] = role
+        envelope.tags.pop("ai.cloud.roleInstance", None)
+        envelope.tags.pop("ai.device.id", None)
+        return envelope
+
+    def _should_collect_stats(self):
+        return False
+
+    def disable_statsbeat(self):
+        """Disable statsbeat for the exporter.
+
+        Setting environment variables is the most complete disable stats, for example,
+        the following function will use environment variables:
+
+        azure.monitor.opentelemetry.exporter.export.logs._exporter._set_statsbeat_custom_events_feature
+
+        But it will also affect the use of metrics in other parts of the program,
+        so we will not call this function to set environment variables temporarily.
+        If we encounter problems in the future, we will solve them when needed.
+        """
+
+        # from azure.monitor.opentelemetry.exporter._constants import _APPLICATIONINSIGHTS_STATSBEAT_DISABLED_ALL
+        # os.environ[_APPLICATIONINSIGHTS_STATSBEAT_DISABLED_ALL] = "false"
```

Comment:
why is this commented?

Reply:
Actually, I'm not sure if I should keep this function. I left this function to indicate a potential problem. If I really encounter statsbeat problems in the future, then this function can quickly identify what problem is.

Reply:
We used to found enable this will lead to extra time (a few seconds) to prepare the telemetry logger. 
If no perf issue now, we can remove this part of code.

https://learn.microsoft.com/en-us/azure/azure-monitor/app/statsbeat?tabs=eu-java%2Cjava

Reply:
There will still be one now, which can be avoided through the **_should_collect_stats** function and **APPLICATIONINSIGHTSSTATSBEAT_DISABLED_ALL**.
Both methods have their advantages and disadvantages. Currently, we will first modify them by modifying the **_should_collect_stats** function

I can delete this extra function first.

![image](https://github.com/microsoft/promptflow/assets/23182548/71700cb6-265b-490d-9b6f-da0f07ca250c)
![image](https://github.com/microsoft/promptflow/assets/23182548/d1f3bda3-b3c0-4e1c-8b77-78f38eb4bdc5)
![image](https://github.com/microsoft/promptflow/assets/23182548/80f35f9c-17e0-4d90-b6a7-dda6f829913c)